src/libopensc/card-authentic.c

```diff
@@ -2305,6 +2306,7 @@ authentic_sm_get_wrapped_apdu(struct sc_card *card, struct sc_apdu *plain, struc
 	rv = card->sm_ctx.module.ops.get_apdus(ctx, &card->sm_ctx.info, NULL, 0, NULL);
 	if (rv < 0) {
 		free(apdu->resp);
+		free((unsigned char *) apdu->data);
```

Comment:
why is the cast needed?

Reply:
The `apdu->data` is `const` so the compiler complains about discarding `const` when passed to `free` without cast.